dashboards_bundlers/server_upload/__init__.py

```diff
@@ -0,0 +1,31 @@
+# Copyright (c) Jupyter Development Team.
+# Distributed under the terms of the Modified BSD License.
+
+import os
+import requests
+from notebook.utils import url_path_join
+from tornado import escape, httpclient
+from tornado.log import access_log
+from ..local_deploy import bundle_file_references
+
+ENDPOINT = '/notebooks/'
+
+def bundle(handler, abs_nb_path):
+    '''
+    Uploads a notebook to a Jupyter Dashboard Server
+    '''
+    # Get name of notebook from filename
+    notebook_basename = os.path.basename(abs_nb_path)
+    notebook_name = os.path.splitext(notebook_basename)[0]
+    if 'DASHBOARD_SERVER_URL' in os.environ:
+        dashboard_url = url_path_join(os.getenv('DASHBOARD_SERVER_URL'), ENDPOINT, escape.url_escape(notebook_name, False))
+        with open(abs_nb_path, 'rb') as notebook:
+            headers = {}
+            if 'DASHBOARD_SERVER_TOKEN' in os.environ:
+                headers['Authorization'] = 'token ' + os.getenv('DASHBOARD_SERVER_TOKEN')
+            result = requests.post(dashboard_url, files={'file': notebook}, headers=headers)
+            result.raise_for_status()
+            handler.redirect(dashboard_url)
+    else:
+        access_log.debug('Can not deploy, DASHBOARD_SERVER_URL not set')
+        raise httpclient.HTTPError(500, message='No dashboard server configured')
```

Comment:
I think you want http://tornado.readthedocs.org/en/latest/web.html#tornado.web.HTTPError, not httpclient error.